provision/initramfs/init

```diff
@@ -70,6 +70,158 @@ for module in `/sbin/detect`; do
 done
 wwsuccess
 
+bondup() {
```

Comment:
In `provision/initramfs/init` there's also a section that starts:

```
if [ -n "$WWPOSTNETDOWN" ]; then
```

Will need to be sure the bond device is handled here, and also the slave devices are taken down when `--postnetdown` is used.

This is how I've generally provisioned a bond device. Provision the node off `eth0` (or whatever), take the network device down when complete, have the image init setup the bond; Use Warewulf files to provision out proper `ifcfg-bond0`, `ifcfg-eth0`, etc... files.

Reply:
in this section the device is read from 

NETDEV=`cat /tmp/wwdev`

my code writes the bond device into that file after successful setup, so --postnetdown should already take it down again. There might be some more nuanced issues with the enslaved devices though, which might also need to be un-configured (by running "ifenslave -d" over all --bonddevs). I'll test that case.

But even in your scenario this patch might already be useful, as assigning multiple hw adresses will help to identify the node when the order of network interfaces changes or the primary interface is actually broken/unglugged. In this case you would *not* configure --bonddevs and init will behave as before (i.e. dont call bondup() at all), just with the added benefit that it doesn't matter if eth0 or eth1 is the primary interface.

Reply:
added code to correctly delete bonding device in case of failure or --postnetdown
busybox's "ip link del" also detaches the slaves automatically